pydantic/_internal/_validate_call.py

```diff
@@ -80,7 +80,15 @@ def __init__(self, function: Callable[..., Any], config: ConfigDict | None, vali
             core_config = config_wrapper.core_config(self)
             schema = _discriminated_union.apply_discriminators(flatten_schema_defs(schema))
             simplified_schema = inline_schema_defs(schema)
-            self.__return_pydantic_validator__ = pydantic_core.SchemaValidator(simplified_schema, core_config)
+            validator = pydantic_core.SchemaValidator(simplified_schema, core_config)
+            if inspect.iscoroutinefunction(self.raw_function):
+
+                async def return_val_wrapper(aw: Awaitable[Any]) -> None:
+                    return validator.validate_python(await aw)
+
+                self.__return_pydantic_validator__ = return_val_wrapper
+            else:
+                self.__return_pydantic_validator__ = validator.validate_python
```

Comment:
We discussed this offline, and the decision was that we'll stick to the implementation currently in this PR, both because it's a smaller diff, a bit easier to understand, and also by not needing to do the `.validate_python` lookup it's likely more performant. 

Still, it was entertaining to see that wrap_validator could be used in this way 😄